internals/api/repo.go

```diff
@@ -40,7 +39,7 @@ type Repo struct {
 
 // Path returns the full repository path.
 func (r Repo) Path() RepoPath {
-	return RepoPath(fmt.Sprintf("%s/%s", r.Owner, r.Name))
+	return RepoPath(r.Owner + "/" + r.Name)
```

Comment:
Consider using the [secretpath](https://godoc.org/github.com/secrethub/secrethub-go/pkg/secretpath) package here. It has convenience functions like `Join` and `AddVersion`. 

Just for your information, the code in this file is a bit old and we don't like it so much. Ideally, we'd move away to something else. I kinda forgot why it's so bad, but I'm sure @SimonBarendse and @jpcoenen have opinions on this.

Reply:
The main problem with the `api` package at the moment is that it is used for the contract between way too many packages.

It's used as an interface between client and server, but it is also used as an interface between the client ([`github.com/secrethub/secrethub-go/pkg/secrethub`](https://pkg.go.dev/github.com/secrethub/secrethub-go/pkg/secrethub?tab=doc)) and consuming applications.

For the path types, we've moved away from using them in the parameters of the client and use primitive types, such as strings instead, because that's easier to understand an d work with (https://github.com/secrethub/secrethub-go/pull/28).